server/src/main/java/org/opensearch/querygroup/LabelingHeader.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.querygroup;
+
+/**
+ * Main enum define various headers introduced by {@link org.opensearch.plugins.LabelingPlugin}s
+ */
+public enum LabelingHeader {
```

Comment:
As @ansjcy mentioned elsewhere, might be good to add `Request` prefix to class names